Bump pylutron-caseta to v0.18.2

swamplynx opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change


- Fixes a race with asyncio.wait_for
- Adds mapping for PalladiomWireFreeShade
- add support wall mounted occupancy sensor device type in latest RA3 firmware. The newer version of the dependency published to PyPi includes a minor change to the __init__.py to include a new device type for wall mounted occupancy sensors.

Changelog:
https://github.com/gurumitts/pylutron-caseta/blob/master/CHANGELOG.md

Diff:
https://github.com/gurumitts/pylutron-caseta/compare/v0.18.1...v0.18.2

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ x] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [ x] The code change is tested and works locally.
- [ x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ x] There is no commented out code in this PR.
- [ x] I have followed the [development checklist][dev-checklist]
- [ x] I have followed the [perfect PR recommendations][perfect-pr]
- [ x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [x ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [x ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ x] Untested files have been added to `.coveragerc`.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

swamplynx committed:
Bump pylutron-caseta to v0.18.2

Minor bump to pylutron-caseta requirement to support wall mounted occupancy sensor device type in latest RA3 firmware.

bdraco commented:
Thanks for bumping the dependency! Please update the PR description to contain at least one (or multiple) of the following:
* A link to the release notes of this package version, and all versions in between.
* A link to the changelog of this package.
* A link to a Git(Hub) diff/compare view from the current version to the bumped version.
This allows us to review upstream changes, which is needed to decide this change is working as intended and/or if we can include it in, for example, a patch release of Home Assistant.

Thanks 👍

bdraco reviewed (changes requested):
Please see above

bdraco commented:
tagging this since there is also a `asyncio.wait_for` race fix in the this release

swamplynx commented:
> Please see above

Updated.
Commit change view: https://github.com/gurumitts/pylutron-caseta/commit/231580d4cb2419e38b9be0b71d1f3e69985026e4

bdraco reviewed (changes requested):
@swamplynx please try again as your link only includes your commit, and is missing the other changes since the last release which will also be included in this bump:

We need one or more of the following:
* A link to the release notes of this package version, and all versions in between.
* A link to the changelog of this package.
* A link to a Git(Hub) diff/compare view from the current version to the bumped version.
This allows us to review upstream changes, which is needed to decide this change is working as intended and/or if we can include it in, for example, a patch release of Home Assistant.

swamplynx commented:
> @swamplynx please try again as your link only includes your commit, and is missing the other changes since the last release which will also be included in this bump:
> 
> We need one or more of the following:
> 
> * A link to the release notes of this package version, and all versions in between.
> * A link to the changelog of this package.
> * A link to a Git(Hub) diff/compare view from the current version to the bumped version.
>   This allows us to review upstream changes, which is needed to decide this change is working as intended and/or if we can include it in, for example, a patch release of Home Assistant.

Sorry. Changelog linked below:
https://github.com/gurumitts/pylutron-caseta/blob/master/CHANGELOG.md

bdraco reviewed (changes requested):
The CI isn't passing

Please check it and address the errors:

https://github.com/home-assistant/core/actions/runs/6104548189/job/16566822821?pr=99789

swamplynx committed:
Update requirements_all.txt for pylutron-caseta 0.18.2

swamplynx committed:
Update requirements_test_all.txt for pylutron-caseta 0.18.2

bdraco commented:
Thanks @swamplynx

bdraco commented:
Tested on all 3 of my systems 👍

Anto79-ops commented:
hey will this address the Caseta errors in my logs that I get about 60% of time only at startups, for example, below. The integration always works, just startups dump these logs (so I never really submitted an issue for it).

`2023-09-02 17:34:10.715 WARNING (MainThread) [homeassistant.config_entries] Config entry 'Lutron' for lutron_caseta integration not ready yet: Timed out while trying to connect to 192.168.1.107; Retrying in background`

```
2023-09-02 17:34:10.743 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/pylutron_caseta/smartbridge.py", line 709, in _login
    await self._subscribe_to_button_status()
  File "/usr/local/lib/python3.11/site-packages/pylutron_caseta/smartbridge.py", line 1244, in _subscribe_to_button_status
    response, _ = await self._subscribe(
                  ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pylutron_caseta/smartbridge.py", line 270, in _subscribe
    raise BridgeDisconnectedError()
pylutron_caseta.BridgeDisconnectedError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/pylutron_caseta/smartbridge.py", line 724, in _login
    self._login_completed.set_exception(ex)
asyncio.exceptions.InvalidStateError: invalid state
2023-09-02 17:34:11.245 ERROR (MainThread) [bellows.uart] Lost serial connection: ConnectionResetError('Remote server closed connection')
2023-09-02 17:34:11.254 ERROR (MainThread) [zigpy.application] Couldn't start application
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/zigpy/application.py", line 195, in startup
    await self.connect()
  File "/usr/local/lib/python3.11/site-packages/bellows/zigbee/application.py", line 141, in connect
    await ezsp.startup_reset()
  File "/usr/local/lib/python3.11/site-packages/bellows/ezsp/__init__.py", line 151, in startup_reset
    await self._gw.wait_for_startup_reset()
  File "/usr/local/lib/python3.11/site-packages/bellows/uart.py", line 175, in wait_for_startup_reset
    await self._startup_reset_future
ConnectionResetError: Remote server closed connection
```


```
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/pylutron_caseta/smartbridge.py", line 709, in _login
    await self._subscribe_to_button_status()
  File "/usr/local/lib/python3.11/site-packages/pylutron_caseta/smartbridge.py", line 1244, in _subscribe_to_button_status
    response, _ = await self._subscribe(
                  ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pylutron_caseta/smartbridge.py", line 270, in _subscribe
    raise BridgeDisconnectedError()
pylutron_caseta.BridgeDisconnectedError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/pylutron_caseta/smartbridge.py", line 724, in _login
    self._login_completed.set_exception(ex)
asyncio.exceptions.InvalidStateError: invalid state
```
